### PR TITLE
dropdown deselection

### PR DESCRIPTION
**Pull request by gabrieleborelli**
In order to achieve the deselection from the Dropdown is used its search input box.
When the search value is fully deleted a **deselect** event is emitted.
Is client responsability to update the **selected** attribute with an empty value in order to reset Dropdown internal state.

Note that this feature is available only for searchable dropdowns.

ref #220

### CODE REVIEW

**Comment by StefanoMagrassi** [comment]
@gabrieleborelli please open a related issue with and add a description of the feature implemented in the PR

File: src/dropdown/index.js [hunk 1]
```diff
@@ -175,6 +183,12 @@ class DropdownClab {
       : null;
   }
 
+  _compIsValidSelection(selected, options, label) {
+    return !isNil(selected) && Array.isArray(options)
+      ? options.filter(o => o[label] === selected[label]).length > 0
```
Comment (StefanoMagrassi) [inline src/dropdown/index.js]: @gabrieleborelli if the validation logic needs at least 1 option selected you could use the `.find` method which exits after the first match (if any):

```js
typeof options.find(o => o[label] === selected[label]) !== 'undefined'
```

Comment (gabrieleborelli) [reply src/dropdown/index.js]: ok

**Comment by gabrieleborelli** [comment]
@StefanoMagrassi, added bare minimum tests for default properties similar to tests already present. I don't know if it's possible to test computed (without moving in libs) or the new event triggered

**Pull request merged**